interfaces/repo.go

```diff
@@ -425,16 +478,15 @@ func (r *Repository) Connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 	r.m.Lock()
 	defer r.m.Unlock()
 
+	return r.connected(snapName, plugOrSlotName)
+}
+
+func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, error) {
 	if snapName == "" {
-		// Look up the core snap if no snap name was given
-		switch {
-		case r.slots["core"] != nil:
-			snapName = "core"
-		case r.slots["ubuntu-core"] != nil:
-			snapName = "ubuntu-core"
-		default:
-			return nil, fmt.Errorf("snap name is empty")
-		}
+		snapName, _ = r.guessCoreSnapName()
+	}
+	if snapName == "" {
```

Comment:
IMO this if could be nested in the previous one, if snapName is not empty it is checked twice.